Docs Restructure: Covering primitives to shared types + Actor->Control-flow + Conversions (#4923)
dfinity/motoko: doc/md/fundamentals/objects-classes.md

        return balance;
    };

    public shared func withdraw(amount: Nat): async ?Nat {
        if (amount > balance) { return null };
        balance -= amount;
        return ?balance;
    };
}
```

## Classes

A class acts as a blueprint for creating multiple objects with independent state.

```motoko no-repl
class Account(initialBalance: Nat) {
    var balance: Nat = initialBalance;

    public func deposit(amount: Nat): async Nat {
        balance += amount;
        balance
    };

    public func withdraw(amount: Nat): ?Nat {
        if (amount > balance) { return null };
        balance -= amount;
        ?balance
    };
};

// Creating multiple accounts
let account1 = Account(500);
let account2 = Account(1000);
```

### Object classes

An object class defines a blueprint for multiple objects.

```motoko no-repl
object class Account(initialBalance: Nat) {
    var balance: Nat = initialBalance;

    public func deposit(amount: Nat): Nat {
        balance += amount;
        balance
    };

    public func withdraw(amount: Nat): ?Nat {
        if (amount > balance) { return null };
        balance -= amount;
        ?balance
    };
}
```

### Module classes

> A module class can act as a factory, producing multiple modules with different configurations.

Reviewer: ```suggestion
A module class can be used to produce multiple modules with different configurations.
```

```motoko no-repl
module class ExchangeRate(baseRate: Float) {
    public func convert(amount: Nat): Float {
        return Float.fromInt(amount) * baseRate;
    };
}

// Creating different currency converters
let usdConverter = ExchangeRate(1.1);
let eurConverter = ExchangeRate(0.9);

Debug.print(Float.toText(usdConverter.convert(100)));  // "110.0"
Debug.print(Float.toText(eurConverter.convert(100)));  // "90.0"
```

### Actor classes

> An actor class allows creating multiple independent actors. This allows dynamic creation of canisters.

Reviewer: ```suggestion
An actor class allows creating multiple independent actors and thus dynamic creation of canisters.
```
